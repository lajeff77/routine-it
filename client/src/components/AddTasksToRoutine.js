import React, {useState} from "react";
import {Box, Button, Card, TextField, Typography} from '@mui/material';
import TaskList from './TaskList';


const AddTasksToRoutine = ({props}) => {

    const [item,setItem] = useState("");

    const handleChange = (e) => {
        setItem(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        props.routineItems.push(item);
        setItem("");
        if(props.routineItems.length > 0)
            props.setReadyForNext(true);
        else
            props.setReadyForNext(false);
    }

    return(
        <Box sx={{margin:"20px", display:"flex", flexDirection:"column", alignItems:"center"}}>
            <Typography variant="h1" sx={{textAlign: 'center', margin: '30px'}}>Let's add some tasks to {props.getRoutineName()}.</Typography>
            <Card variant="outlined" sx={{align:"center", width:"70%", display:"flex", flexDirection:"column", outline:"2px"}}>
                <Typography variant="body1"  sx={{margin: '20px'}}>Don't hold back! What is everything you ideally want to accomplish with {props.getRoutineName()}?</Typography>
                <Box component="form" onSubmit={handleSubmit} sx={{margin:"20px", display:"flex", flexDirection:"row", justifyContent:"center", alignItems:"center"}}>
                    <TextField required value={item} onChange={handleChange} id="outlined-required" label="Enter a new task" variant="outlined" sx={{margin: '20px'}}/>
                    <Button variant="contained" type="submit" sx={{width:"100px", margin:"10px", padding:"10px"}}>Add</Button>
                </Box> 
                <TaskList props={{list: props.routineItems, setList: props.setRoutineItems}}/>
            </Card>
        </Box>
    )
}

export default AddTasksToRoutine;