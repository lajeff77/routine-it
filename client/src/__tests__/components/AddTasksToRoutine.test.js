import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import AddTasksToRoutine from "../../components/AddTasksToRoutine";


const testProps  = {
    getRoutineName: () => {return "My Routine"},
    routineItems: ["Eat breakfast", "Walk dog", "Clean dishes", "Shower"]
}

describe(AddTasksToRoutine.name, ()=>{
    test('AddTasksToRoutine matches snapshot', () => {
        const {container} = render(<AddTasksToRoutine props={testProps}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});