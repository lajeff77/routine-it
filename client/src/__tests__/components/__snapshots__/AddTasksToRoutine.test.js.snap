// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTasksToRoutine AddTasksToRoutine matches snapshot 1`] = `
<div
  class="MuiBox-root css-2u9sdq"
>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-3ns5jg-MuiTypography-root"
  >
    Let's add some tasks to 
    .
  </h1>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1yf7ndb-MuiPaper-root-MuiCard-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-altc5y-MuiTypography-root"
    >
      Don't hold back! What is everything you ideally want to accomplish with 
      ?
    </p>
    <form
      class="MuiBox-root css-rsoibw"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-166k1z2-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="outlined-required"
          id="outlined-required-label"
        >
          Enter a new task
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-required"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Enter a new task
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-146d4n0-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
    <div
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;
